python/tk_multi_workfiles/entity_models/extended_model.py

```diff
@@ -0,0 +1,280 @@
+                       the model.
+        """
+        # Please note that some of these members already exist as private members
+        # in the base implementation without any accessor.
+        self._entity_type = entity_type
+        self._original_filters = filters
+        self._hierarchy = hierarchy
+        self._fields = fields
+        self._extra_filters = None
+        self._entity_types = set()
+        super(ShotgunExtendedEntityModel, self).__init__(
+            entity_type,
+            filters,
+            hierarchy,
+            fields,
+            *args,
+            **kwargs
+        )
+
+    @property
+    def represents_tasks(self):
+        """
+        :returns: True if this model represents Tasks.
+        """
+        return self.get_entity_type() == "Task"
+
+    @property
+    def supports_step_filtering(self):
+        """
+        :returns: True if Step filtering can be used with this model
+        """
+        # If don't have steps in the fields we query from Shotgun, we assume
+        # step filtering should be disabled.
+        return "step" in self._fields or "step" in self._hierarchy
+
+    def load_and_refresh(self, extra_filters=None):
+        """
+        Load the data for this model and post a refresh.
+
+        :param extra_filters: A list of additional Shotgun filters which are added
+                              to the initial filters.
+        """
+        self._extra_filters = extra_filters
+        filters = self._original_filters[:] # Copy the list to not update the reference
+        if extra_filters:
+            filters.append(extra_filters)
+        self._load_data(
+            self._entity_type,
+            filters,
+            self._hierarchy,
+            self._fields
+        )
+        self.async_refresh()
+
+    def update_filters(self, extra_filters):
+        """
+        Update the filters used by this model.
+
+        A full refresh is triggered by the update if not using deferred queries.
+        Otherwise, the filter is applied to all expanded items in the model which
+        are direct parent of deferred results.
+
+        :param extra_filters: A list of additional Shotgun filters which are added
+                              to the initial filters.
+        """
+        self._extra_filters = extra_filters
+        filters = self._original_filters[:] # Copy the list to not update the reference
+        if extra_filters:
+            filters.append(extra_filters)
+        self._load_data(
+            self._entity_type,
+            filters,
+            self._hierarchy,
+            self._fields
+        )
+        self.async_refresh()
+
+    def _finalize_item(self, item):
+        """
+        Called every time an item was added in the model.
+        """
+        super(ShotgunExtendedEntityModel, self)._finalize_item(item)
+        # We need to keep track of which entities are in the model, so we can bail
```

Comment:
this comment might be better to move/(or copy) to the ctor where we first introduce the variable